src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -169,6 +188,24 @@ protected ConnectionBufferPool WebSocketBufferPool
             get { return _bufferPool; }
         }
 
+        HashAlgorithm HashAlgorithm
+        {
+            [SecurityCritical]
+            get
+            {
+                if (_hashAlgorithm == null)
+                {
+                    _hashAlgorithm = SHA512.Create();
+                }
+                else
+                {
+                    _hashAlgorithm.Initialize();
+                }
```

Comment:
The original code called Initialize before handing it out each time. With hash algorithms, you maintain a state that is modified/transformed with each new piece of data that you add to the hash. When you are done, there's a final transform done to prevent attacks based on appending data, in other words finalizes the hash such that no more data can be added. Initialize resets this state data. The docs don't say that this is needed each time, but looking at the implementation, it certainly looks like the state isn't implicitly reset each time and that you would end up with an invalid hash if this isn't done. Btw, desktop uses sha1, I switched to SHA512 so that it should hopefully be a decade before we get told we need to stop using this hash algorithm. There's no back compat risks as it's only used in-process and not persisted/transmitted.